x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/MatchPhrase.java

```diff
@@ -0,0 +1,345 @@
+                    description = "Indicates whether all documents or none are returned if the analyzer removes all tokens, such as "
+                        + "when using a stop filter. Defaults to none."
+                ),
+                @MapParam.MapParamEntry(
+                    name = "boost",
+                    type = "float",
+                    valueHint = { "2.5" },
+                    description = "Floating point number used to decrease or increase the relevance scores of the query. Defaults to 1.0."
+                ) },
+            description = "(Optional) MatchPhrase additional options as <<esql-function-named-params,function named parameters>>."
+                + " See <<query-dsl-match-query-phrase,match_phrase query>> for more information.",
+            optional = true
+        ) Expression options
+    ) {
+        this(source, field, matchPhraseQuery, options, null);
+    }
+
+    public MatchPhrase(Source source, Expression field, Expression matchPhraseQuery, Expression options, QueryBuilder queryBuilder) {
+        super(
+            source,
+            matchPhraseQuery,
+            options == null ? List.of(field, matchPhraseQuery) : List.of(field, matchPhraseQuery, options),
+            queryBuilder
+        );
+        this.field = field;
+        this.options = options;
+    }
+
+    @Override
+    public String getWriteableName() {
+        return ENTRY.name;
+    }
+
+    @Override
+    public String functionName() {
+        return ENTRY.name;
+    }
+
+    private static MatchPhrase readFrom(StreamInput in) throws IOException {
+        Source source = Source.readFrom((PlanStreamInput) in);
+        Expression field = in.readNamedWriteable(Expression.class);
+        Expression query = in.readNamedWriteable(Expression.class);
+        QueryBuilder queryBuilder = null;
+        queryBuilder = in.readOptionalNamedWriteable(QueryBuilder.class);
+        return new MatchPhrase(source, field, query, null, queryBuilder);
+    }
+
+    @Override
+    public final void writeTo(StreamOutput out) throws IOException {
+        source().writeTo(out);
+        out.writeNamedWriteable(field());
+        out.writeNamedWriteable(query());
+        out.writeOptionalNamedWriteable(queryBuilder());
+    }
+
+    @Override
+    protected TypeResolution resolveParams() {
+        return resolveField().and(resolveQuery()).and(resolveOptions(options(), THIRD)).and(checkParamCompatibility());
+    }
+
+    private TypeResolution resolveField() {
+        return isNotNull(field, sourceText(), FIRST).and(isType(field, FIELD_DATA_TYPES::contains, sourceText(), FIRST, "keyword, text"));
+    }
+
+    private TypeResolution resolveQuery() {
+        return isType(query(), QUERY_DATA_TYPES::contains, sourceText(), SECOND, "keyword, text").and(
+            isNotNullAndFoldable(query(), sourceText(), SECOND)
+        );
+    }
+
+    private TypeResolution checkParamCompatibility() {
```

Comment:
I think we can remove this method now - params should be compatible as they are text / keyword, no need to double check like we did in match()